docs/modules/ROOT/pages/introspection.adoc

```diff
@@ -1,160 +1,68 @@
-
-== Table of Contents
-
-* <<erc165,ERC165>>
- ** <<interface_calculations,Interface calculations>>
- ** <<registering_interfaces,Registering interfaces>>
- ** <<querying_interfaces,Querying interfaces>>
- ** <<ierc165,IERC165>>
- ** <<ierc165_api_specification,IERC165 API Specification>>
-  *** <<supportsinterface,`supportsInterface`>>
- ** <<erc165_library_functions,ERC165 Library Functions>>
-  *** <<supportsinterface2,`supports_interface`>>
-  *** <<register_interface,`register_interface`>>
-
-== ERC165
-
-The ERC165 standard allows smart contracts to exercise https://en.wikipedia.org/wiki/Type_introspection[type introspection] on other contracts, that is, examining which functions can be called on them.
-This is usually referred to as a contract's interface.
-
-Cairo contracts, like Ethereum contracts, have no native concept of an interface, so applications must usually simply trust they are not making an incorrect call.
-For trusted setups this is a non-issue, but often unknown and untrusted third-party addresses need to be interacted with.
-There may even not be any direct calls to them!
-(e.g.
-ERC20 tokens may be sent to a contract that lacks a way to transfer them out of it, locking them forever).
-In these cases, a contract declaring its interface can be very helpful in preventing errors.
+:eip165: https://eips.ethereum.org/EIPS/eip-165[EIP165]
+:src5: https://github.com/starknet-io/SNIPs/blob/main/SNIPS/snip-5.md[SRC5]
+:src5-rs: https://github.com/ericnordelo/src5-rs[src5-rs]
 
-It should be noted that the https://github.com/OpenZeppelin/cairo-contracts/blob/release-v0.6.1/src/openzeppelin/utils/constants/library.cairo[constants library] includes constant variables referencing all of the interface ids used in these contracts.
-This allows for more legible code i.e.
-using `IERC165_ID` instead of `0x01ffc9a7`.
+= Introspection
 
-=== Interface calculations
+In the Ethereum ecosystem, smart contracts often need to exercise https://en.wikipedia.org/wiki/Type_introspection[type introspection]
+on other contracts, for interoperability purposes. The {eip165} standard supports this allowing contracts to declare
+their support for a given interface, and for other contracts to query this support.
 
-In order to ensure EVM/StarkNet compatibility, interface identifiers are not calculated on StarkNet.
-Instead, the interface IDs are hardcoded from their EVM calculations.
-On the EVM, the interface ID is calculated from the selector's first four bytes of the hash of the function's signature while Cairo selectors are 252 bytes long.
-Due to this difference, hardcoding EVM's already-calculated interface IDs is the most consistent approach to both follow the EIP165 standard and EVM compatibility.
+Starknet is not an exception to this rule, having to exercise type introspection among smart contracts as well. From the
+differences between Starknet and Ethereum, instead of trying to replicate the EVM's interface ID calculation, the community
```

Comment:
```suggestion
Instead of trying to replicate the EVM's interface ID calculation, the community
```

Not a huge deal, but cases like this are why I prefer one whole sentence per line. It's still better than one paragraph per line :)